### ISSUE TITLE
The sorting elements in the offline list don't work.

### ISSUE THREAD

**Issue by mostafaalagamy** [Malopieds/InnerTune#142]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### 

### Expected behavior

Sorting should be done according to the selection.

### Actual behavior

Nothing happens

### Screenshots/Screen recordings

![Screenshot_2024-06-27-10-55-20-847_com metrolist music](https://github.com/Malopieds/InnerTune/assets/80542861/5b4e487d-8615-4137-8f1b-b29145f92ca5)

### 

### InnerTune version

debug version

### Android version

13

**Comment by Malopieds** [comment]
Yep, the more songs you have, the more time it will take to find downloaded songs (and sort them accordingly). What I was thinking is create a "real" playlist of downloaded song. The only cons of this would be that it will be exported with the backup (but not downloaded directly). What do you think?

**Comment by mostafaalagamy** [comment]
> Yep, the more songs you have, the more time it will take to find downloaded songs (and sort them accordingly). What I was thinking is create a "real" playlist of downloaded song. The only cons of this would be that it will be exported with the backup (but not downloaded directly). What do you think?

The suggested idea would be great for remembering the downloaded songs when restoring the backup, it's a good idea.

**Comment by mostafaalagamy** [comment]
Also Sort by date added in likes Not working !!

**Comment by Malopieds** [comment]
In liked playlist and liked songs it works for me, are you using the latest debug version?

**Comment by mostafaalagamy** [comment]
> In liked playlist and liked songs it works for me, are you using the latest debug version?

Yep latest , Look at this .

https://github.com/user-attachments/assets/f1bb4b0a-e51c-4cf8-8d84-9e2d1fe29b2c

**Comment by mostafaalagamy** [comment]
For your information, the 'Sort by Date added' works in the 'Liked' of the 'Song Filter' in library.

**Comment by kaneryu** [comment]
> Yep, the more songs you have, the more time it will take to find downloaded songs (and sort them accordingly). What I was thinking is create a "real" playlist of downloaded song. The only cons of this would be that it will be exported with the backup (but not downloaded directly). What do you think?

This should be the default behavior

**Comment by fbbb2** [comment]
I have this problem too. Nothing happens when i want to sort the downloads.
